### ISSUE TITLE
Moving to firebase functions

### ISSUE THREAD

**Issue by Andrewnt219** [Andrewnt219/rent-near-me#127]
Still deciding if we should do it or not. From what I have tested on [seneca-course-reviews](https://github.com/Andrewnt219/seneca-course-reviews), these are the pros and cons on a scale of 0 to 3.

Pros
- Have a monorepo to separate backend and frontend (+3)
- Multiple trips between backend and DB should theoretically be faster (+1).
- We will have firebase functions code eventually for functions like `document.onWrite` (+1).
- Don't need to include firebase secrets inside repo. This is a must-have if we are open to more contributors (+2).

Cons
- Cold boot is slower (4-4.5s on average vs Next.js 3s). This should not be a problem if the web app is popular and used frequently though (-2).
- No absolute import in backend (-1)

**Comment by vitokhangnguyen** [comment]
So I assume we decided to scratch this idea?

**Comment by Andrewnt219** [comment]
yeah, not worth the refactor. Still considering separated backend repo.
